src/mqt/yaqs/characterization/memory/backends/surrogates/utils.py

```
@@ -0,0 +1,290 @@
+
+    def __call__(self, rho: np.ndarray) -> np.ndarray:
+        """Apply the rank-1 intervention map to a single-qubit density matrix.
+
+        Returns:
+            Updated single-qubit density matrix after the rank-1 map.
+        """
+        r = np.asarray(rho, dtype=np.complex128).reshape(2, 2)
+        return np.trace(self.effect @ r) * self.rho_prep
+
+
+def _initial_mcwf_state_from_rho0(
```

Ruff (0.15.18)

[warning] Too many branches (21 > 12)

(PLR0912)